contracts/utils/cryptography/P256.sol

```diff
@@ -242,10 +274,16 @@ library P256 {
      * @dev Compute G·u1 + P·u2 using the precomputed points for G and P (see {_preComputeJacobianPoints}).
      *
      * Uses Strauss Shamir trick for EC multiplication
-     * https://stackoverflow.com/questions/50993471/ec-scalar-multiplication-with-strauss-shamir-method.
-     * We optimise on this a bit to do with 2 bits at a time rather than a single bit.
-     * The individual points for a single pass are precomputed.
-     * Overall this reduces the number of additions while keeping the same number of doublings.
+     * https://stackoverflow.com/questions/50993471/ec-scalar-multiplication-with-strauss-shamir-method
+     *
+     * We optimise on this a bit to do with 2 bits at a time rather than a single bit. The individual points for a
+     * single pass are precomputed. Overall this reduces the number of additions while keeping the same number of
+     * doublings
+     *
+     * Note that some particular values, such as P = -G (private key = N-1) will result in invalid point computation.
+     * This would result in the inability to verify or recover valid signatures. This edge case could potentially
+     * cause a valid signature to fail verification. While specific private keys are known to trigger that issue, is
+     * it extremely unlikelly to happen with randomly generated keys.
```

Comment:
```suggestion
     * https://stackoverflow.com/questions/50993471/ec-scalar-multiplication-with-strauss-shamir-method
     *
     * We optimize this for 2 bits at a time rather than a single bit. The individual points for a single pass are 
     * precomputed. Overall this reduces the number of additions while keeping the same number of
     * doublings
     *
     * Note that some particular values, such as P = -G (private key = N-1) will result in invalid point computation.
     * This would result in the inability to verify or recover valid signatures. This edge case could potentially
     * cause a valid signature to fail verification. While specific private keys are known to trigger that issue, it
     * is almost impossible to happen with randomly generated keys.
```